beacon-chain/core/blocks/benchmarks_test.go

```diff
@@ -0,0 +1,282 @@
+		b.Fatal(err)
+	}
+	root, err := ssz.HashTreeRoot(s)
+	if err != nil {
+		b.Fatal(err)
+	}
+	block.StateRoot = root[:]
+	blockRoot, err := ssz.SigningRoot(block)
+	if err != nil {
+		b.Fatal(err)
+	}
+	// Temporarily incrementing the beacon state slot here since BeaconProposerIndex is a
+	// function deterministic on beacon state slot.
+	beaconState.Slot++
+	proposerIdx, err := helpers.BeaconProposerIndex(beaconState)
+	if err != nil {
+		b.Fatal(err)
+	}
+	beaconState.Slot--
+	domain := helpers.Domain(beaconState.Fork, helpers.CurrentEpoch(beaconState), params.BeaconConfig().DomainBeaconProposer)
+	block.Signature = privs[proposerIdx].Sign(blockRoot[:], domain).Marshal()
+
+	blockSSZ, err := ssz.Marshal(block)
+	if err != nil {
+		b.Fatal(err)
+	}
+	if err = ioutil.WriteFile("128Attblock.ssz", blockSSZ, 0644); err != nil {
+		b.Fatal(err)
+	}
+
+	cleanStates := createCleanStates(beaconState)
+
+	fmt.Println("states generated")
+
+	b.N = runAmount
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		fmt.Println(i)
+		if _, err := state.ExecuteStateTransition(context.Background(), cleanStates[i], block); err != nil {
+			b.Fatal(err)
+		}
+	}
+}
+
+func BenchmarkExecuteStateTransition_ReadSaved(b *testing.B) {
+	slotsPerEpoch := params.BeaconConfig().SlotsPerEpoch
+	committeeSize := (uint64(validatorCount) / slotsPerEpoch) / (benchmarkConfig().MaxAttestations / slotsPerEpoch)
+	c := params.BeaconConfig()
+	c.PersistentCommitteePeriod = 0
+	c.MinValidatorWithdrawabilityDelay = 0
+	c.TargetCommitteeSize = committeeSize
+	c.MaxAttestations = benchmarkConfig().MaxAttestations
+	params.OverrideBeaconConfig(c)
+	defer params.OverrideBeaconConfig(params.MainnetConfig())
+
+	beaconState := createBeaconState(b)
+
+	privs, _, err := interop.DeterministicallyGenerateKeys(0, uint64(validatorCount))
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	conf := &testutil.BlockGenConfig{
+		MaxAttestations: 0,
+	}
+
+	blockBytes, err := ioutil.ReadFile("127Attblock.ssz")
+	if err != nil {
+		b.Fatal(err)
+	}
+	block := &ethpb.BeaconBlock{}
+	if err := ssz.Unmarshal(blockBytes, block); err != nil {
+		b.Fatal(err)
+	}
+
+	// Process beacon state to mid-epoch to prevent epoch calculations from manipulating benchmarks.
+	for i := uint64(0); i < 6+params.BeaconConfig().SlotsPerEpoch-1; i++ {
+		fmt.Printf("state at slot %d\n", beaconState.Slot)
+		block := testutil.GenerateFullBlock(b, beaconState, privs, conf)
+		beaconState, err = state.ExecuteStateTransitionNoVerify(context.Background(), beaconState, block)
+		if err != nil {
+			b.Error(err)
+		}
+	}
+	cleanStates := createCleanStates(beaconState)
+
+	fmt.Println("states generated")
+
+	b.N = runAmount
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		fmt.Println(i)
+		if _, err := state.ExecuteStateTransition(context.Background(), cleanStates[i], block); err != nil {
+			b.Fatal(err)
+		}
+	}
+}
+
+func BenchmarkHashTreeRoot_65536Validators(b *testing.B) {
+	slotsPerEpoch := params.BeaconConfig().SlotsPerEpoch
```

Comment:
this whole chunk of setting up to run a transition can be abstracted into a separate method.